### PR TITLE
fix windows path (#660)

### PR DESCRIPTION
**Pull request by parthban-db**
## Changes

Changed `pathlib.Path` with the `pathlib.PurePosixPath` in `/databricks/sdk/mixins/files.py` which always use linux path separators regardless of the OS that it is running on. Fixes (#660)

## Tests

- [x] `make test` run locally
- [x] `make fmt` applied
- [ ] relevant integration tests applied

### CODE REVIEW

File: databricks/sdk/mixins/files.py [hunk 1]
```diff
@@ -267,7 +267,8 @@ def __repr__(self) -> str:
 class _Path(ABC):
 
     def __init__(self, path: str):
-        self._path = pathlib.Path(str(path).replace('dbfs:', '').replace('file:', ''))
+        self._path = pathlib.PurePosixPath(str(path).replace('dbfs:', '').replace('file:', ''))
+        self._system_path = pathlib.Path(self._path)
```
Comment (edwardfeng-db) [inline databricks/sdk/mixins/files.py:271]: It'd be good to add a comment explaining why we need `self._path` to be `PurePosixPath` and why we need `self._system_path`

**Comment by edwardfeng-db** [comment]
Can you link the github issue that this PR solves in the PR description?

**Review by edwardfeng-db — APPROVED** [review]
LGTM, let's get @mgyucht to take a look before merging

**Review by mgyucht — CHANGES_REQUESTED** [review]
Thank you for your contribution @parthban-db! A couple notes about this PR:

The more I think about this, the more I think we're putting too much implementation detail in one place. I would do the following:
1. Change `_Path.__init__` to be an `@abstractmethod` and remove its implementation.
2. Add `__init__` implementations in each of `_DbfsPath`, `_LocalPath` and `_VolumesPath`. In `_DbfsPath` and `_VolumesPath`, use `PurePosixPath`, and in `_LocalPath` use `Path`.
This way, we don't duplicate the `path` state, and each implementation can choose the specific path type needed for itself.

Second: We should also make sure we add a test case that covers the actual issue reported by the user: when making a directory in DBFS, regardless of the OS, the path separator used in the API call must be `/`, not `\`. 

@edwardfeng-db could you help @parthban-db modify the unit test runners to support testing in Windows as well?

File: tests/test_dbfs_mixins.py [hunk 2]
```diff
@@ -70,3 +70,22 @@ def test_fs_path_invalid(config):
...
+
+def test_dbfs_local_path_mkdir(config, tmp_path):
+    from databricks.sdk import WorkspaceClient
+
+    w = WorkspaceClient(config=config)
+    w.dbfs._path(f'file:{tmp_path}/test_dir').mkdir()
+    assert w.dbfs.exists(f'file:{tmp_path}/test_dir')
+
+
+def test_dbfs_exists(config, mocker):
```
Comment (mgyucht) [inline tests/test_dbfs_mixins.py:83]: Can we do this test for volumes as well?

**Review by mgyucht — COMMENTED** [review]
LGTM once you add the volumes test

**Review by mgyucht — APPROVED** [review]

**Pull request merged**